tla-io/src/main/scala/at/forsyte/apalache/tla/imp/SanyImporter.scala

```diff
@@ -106,34 +106,41 @@ class SanyImporter(sourceStore: SourceStore, annotationStore: AnnotationStore) e
    * into it, in order to call SANY.
    *
    * @param source
-   *   the text source
+   *   the text source for the root module
+   * @param aux
+   *   the text sources for any auxiliary modules
    * @return
    *   the pair (the root module name, a map of modules)
    */
-  def loadFromSource(source: Source): (String, Map[String, TlaModule]) = {
+  def loadFromSource(source: Source, aux: Seq[Source] = Seq()): (String, Map[String, TlaModule]) = {
+    val tempDir = Files.createTempDirectory("sanyimp").toFile
+    try {
+      val rootModule = saveTlaFile(tempDir, source)
+      aux.foreach(saveTlaFile(tempDir, _))
+      loadFromFile(rootModule)
+    } finally {
+      tempDir.delete()
+    }
+  }
+
+  private def saveTlaFile(dir: File, source: Source): File = {
     val moduleName = moduleNameOfSource(source) match {
       case Some(n) => n
-      case None    => throw new SanyImporterException(s"No module name found in given source")
+      case None    => throw new SanyImporterException(s"No module name found in source for module")
```

Comment:
Can we pass the file / module name here? Otherwise it might be hard to debug.

Reply:
Maybe there should be a test for that.

Reply:
I added the first line of the source to the error message in fe66642038af698a4809607988860bb653d40f8d